cdap-app-templates/cdap-program-report/cdap-program-report-base/src/main/java/co/cask/cdap/report/main/ProgramStartInfo.java

```diff
@@ -27,11 +28,15 @@
   private Map<String, String> runtimeArguments;
   private ArtifactId artifactId;
   private String principal;
+  @Nullable
+  private String triggeringScheduleInfo;
 
-  public ProgramStartInfo(Map<String, String> arguments, ArtifactId artifactId, String principal) {
+  public ProgramStartInfo(Map<String, String> arguments, ArtifactId artifactId, String principal,
+                          @Nullable String scheduleInfo) {
```

Comment:
It's better to pass in the actual object here and return the actual object instead of a string. Though I guess you're not doing that because the serialization becomes really weird?

Maybe it would make sense to pass in all the system arguments then? I don't think storing the stringified schedule info is a good idea, it makes this code super brittle.

Reply:
yes, serialization becomes weird, and felt unnecessary, as its going to be written to file and not really used by the code that serializes, however it might be better to store the entire system runtime arguments, as you mentioned to support future use cases if any, without breaking compat. ill make that change.